### PR TITLE
Pull request for validation tools

### PR DESCRIPTION
**Pull request by DennRoy**
Hi Jan,

I made some improvements to the validation tools. Could you take a look at it if you have the time?

Cheers,
Dennis

### CODE REVIEW

File: compare.py [hunk 1]
```diff
@@ -58,7 +62,7 @@ def overlay(hists, ytitle, header, addon):
         hist.SetLineWidth(2)
         hist.SetMarkerSize(1)
 
-        for ip in xrange(hist.GetN()):
+        for ip in xrange(0,hist.GetN()):
```
Comment (steggema) [inline compare.py]: This change is not needed

File: compare.py [hunk 2]
```diff
@@ -95,14 +104,41 @@ def overlay(hists, ytitle, header, addon):
...
+    for alpha,beta in directory2options.items():
+	if alpha in header: directory2 = beta
+    for alpha,beta in directory22options.items():
+	if alpha in header: directory2 = beta
+
+    if "_eta" in header:
+	if directory1 != "": directory1 += "_eta"
+	if directory2 != "": directory2 += "_eta"
+
+    if directory1 != "": save(canvas, 'compare_' + runtype + '/' + directory1 + '/' + header)
```
Comment (steggema) [inline compare.py]: if directory1: 
# via pep8
# same for following lines

File: compare.py [hunk 3]
```diff
@@ -233,16 +272,22 @@ def makeEffPlotsVars(tree, varx, vary, sel, nbinx, xmin, xmax, nbiny, ymin, ymax
     parser = argparse.ArgumentParser(formatter_class=argparse.ArgumentDefaultsHelpFormatter)
 
     parser.add_argument('runtype', choices=['ZTT', 'ZEE', 'ZMM', 'QCD', 'TTbar', 'TTbarTau', 'ZpTT'], help='choose sample type')
-    parser.add_argument('--releases', help='comma separated list of releases', default='CMSSW_9_4_0_pre1,CMSSW_9_4_0_pre2')
+    parser.add_argument('-r', '--releases', help='comma separated list of releases', default='CMSSW_9_4_0_pre1,CMSSW_9_4_0_pre2')
+    parser.add_argument('-g', '--globaltag', help='comma separated global tags of releases', default='93X_mc2017_realistic_v3-v1,94X_mc2017_realistic_v1-v1')
+    parser.add_argument('-p', '--part',  help='Make WP plots(1), first half of histogram plots(2), second half of histogram plots(3), or everything at once(0) (This part needs to be split up to avoid a crash that happens for some reason)', default=0)
```
Comment (steggema) [inline compare.py:277]: I hope we can get rid of this part thing in the future but let's be practical.

**Review by steggema — COMMENTED** [review]
Looks good, just a few minor style changes that you can implement if they're fine for you

**Comment by DennRoy** [comment]
Thanks for your comments! I implemented the changes.

**Pull request merged**